eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryImpl.java

```diff
@@ -111,12 +122,108 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantEn
             exclude.forEach(e -> query.fields().exclude(e));
         }
 
-        return mongoTemplate.find(query, VariantEntity.class);
+        List<VariantEntity> variantEntities = mongoTemplate.find(query, VariantEntity.class);
+
+        Map<String, Map<String, Map<String, String>>> studyIdsToFileIdsToIndexesToNames =
+                studyIdsToFileIdsToIndexesToNames();
+
+        variantEntities = updateVariantEntitiesSampleNames(variantEntities, studyIdsToFileIdsToIndexesToNames);
+
+        return variantEntities;
+    }
+
+    private Map<
+            String, Map<
+            String, Map<
+            String, String>>> studyIdsToFileIdsToIndexesToNames() {
```

Comment:
The 3-level `Map` could be replaced by a Guava `Table`, which has the same underlying implementation but a much nicer interface: https://github.com/google/guava/wiki/NewCollectionTypesExplained#table